spec/models/project_spec.rb

```diff
@@ -3,6 +3,307 @@
+        csv_string << "9,Story title,feature,#{user.name},#{user.name},description,Task 1,completed"
+
+        project.stories.from_csv csv_string
+        expect(tasks.count).to eq(1)
+      end
+
+      context 'when task_status not_completed' do
+        let(:not_completed_task_csv) do
+          "9,Story title,feature,#{user.name},#{user.name},description,Task 1,not_completed"
+        end
+
+        it 'task.done is false' do
+          csv_string << not_completed_task_csv
+
+          project.stories.from_csv csv_string
+          expect(task.done).to be false
+        end
+      end
+
+      context 'when task_status completed' do
+        let(:completed_task_csv) do
+          "9,Story title,feature,#{user.name},#{user.name},description,Task 1,completed"
+        end
+
+        it 'task.done is true' do
+          csv_string << completed_task_csv
+
+          project.stories.from_csv csv_string
+          expect(task.done).to be true
+        end
+      end
+    end
+  end
+
+  describe '#csv_filename' do
+    subject { build(:project, name: 'Test Project') }
+
+    its(:csv_filename) { should match(/^Test Project-\d{8}_\d{4}\.csv$/) }
+  end
+
+  describe '.archived' do
+    let(:normal_project) { create :project }
+    let(:archived_project) do
+      create :project,
+      archived_at: Time.current
+    end
+    subject { described_class.archived }
+
+    it 'includes archived projects' do
+      expect(subject).to include archived_project
+    end
+
+    it 'excludes non-archived projects' do
+      expect(subject).not_to include normal_project
+    end
+  end
+
+  describe '#iteration_service' do
+    let(:iteration_service_stub) { instance_double('Central::Support::IterationService') }
+
+    context 'when since and current_time are defined' do
+      let(:since) { Time.current - 1.day }
+      let(:current_time) { Time.current }
+
+      before do
+        allow(Central::Support::IterationService)
+          .to receive(:new)
+          .with(subject, since: since, current_time: current_time)
+          .and_return(iteration_service_stub)
+      end
+
+      it 'uses them to initialize Central::Support::IterationService' do
+        expect(subject.iteration_service(since: since, current_time: current_time))
+          .to eq iteration_service_stub
+      end
+    end
+
+    context 'when since and current_time are not defined' do
+      let(:current_time) { Time.current }
+
+      before do
+        Timecop.freeze(current_time) do
+          allow(Central::Support::IterationService)
+            .to receive(:new)
+            .with(subject, since: nil, current_time: current_time)
+            .and_return(iteration_service_stub)
+        end
+      end
+
+      it 'defaults nil to since and the current time to current_time' do
+        Timecop.freeze(current_time) do
+          expect(subject.iteration_service).to eq iteration_service_stub
+        end
+      end
+    end
+  end
+
+  subject { build :project }
```

Comment:
Should this be here?

Reply:
It was moved to the right place, thanks!